okta-direct-auth/src/androidMain/kotlin/com/okta/directauth/DirectAuthenticationFlowBuilder.kt

```diff
@@ -0,0 +1,146 @@
+
+    /**
+     * The clock used for time-sensitive operations.
+     *
+     * This defaults to a clock that returns the current time in epoch seconds.
+     */
+    var clock: OidcClock = OidcClock { System.currentTimeMillis() / 1000 }
+
+    /**
+     * Any additional query string parameters you would like to supply to the authorization server for all requests from this flow.
+     *
+     * Defaults to an empty map.
+     */
+    var additionalParameter: Map<String, String> = emptyMap()
+
+    companion object {
+        /**
+         * Creates an instance of [DirectAuthenticationFlow] using the builder pattern.
+         *
+         * @param issuerUrl The base URL of the Authorization Server. This is the issuer URI for the authorization server that will be used for the flow. For example: `https://dev-123456.okta.com`.
+         * @param clientId The client ID of the application. This ID is obtained from the Okta developer console when you register your application.
+         * @param scope The OAuth 2.0 scopes the application is requesting. Scopes are used to specify what access privileges are being requested for access tokens. For example: `openid`, `profile`, `email`, and `offline_access`.
+         * @param buildAction A lambda with a [DirectAuthenticationFlowBuilder] receiver to configure
+         * the flow's parameters.
+         * @return A [Result] containing the configured [DirectAuthenticationFlow] on success,
+         * or an exception on failure.
+         */
+        fun create(
+            issuerUrl: String,
+            clientId: String,
+            scope: List<String>,
+            buildAction: (DirectAuthenticationFlowBuilder.() -> Unit)? = null,
+        ): Result<DirectAuthenticationFlow> = runCatching {
+            val builder = DirectAuthenticationFlowBuilder()
+            buildAction?.invoke(builder)
+
+            require(runCatching {
+                val url = Url(issuerUrl)
+                url.protocol == URLProtocol.HTTPS && url.host.isNotBlank()
+            }.getOrDefault(false)) { "issuerUrl must be a valid https URL." }
+
+            require(clientId.isNotBlank()) { "clientId must be set and not empty." }
+            require(scope.isNotEmpty()) { "scope must be set and not empty." }
+
+            Result.success(
+                DirectAuthenticationFlowImpl(
+                    DirectAuthenticationContext(
+                        issuerUrl,
+                        clientId,
+                        scope,
+                        builder.clientSecret,
+                        builder.supportedGrantType,
+                        builder.acrValues,
+                        builder.directAuthenticationIntent,
+                        builder.apiExecutor,
+                        builder.logger,
+                        builder.clock,
+                        builder.additionalParameter,
+                    )
+                )
+            )
+        }.getOrElse { Result.failure(it) }
+    }
+}
```

Comment:
Missing newline. Could you add that here? Not sure how spotless didn't catch it

Reply:
I will need to setup spotless for this project. I have not yet because of the constant changes. This will be done soon as a clean up.